src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import usePortal from 'react-cool-portal';
+import styled from 'styled-components';
+import { isNotNull, isNotUndefined } from 'ramda-adjunct';
+
+import { ModalProps } from './Modal.types';
+import { ModalWidthSizes } from './Modal.enums';
+import { useCalculateContentHeight } from './hooks/useCalculateContentHeight';
+import { useLockBodyScroll } from './hooks/useLockBodyScroll';
+import { useOuterClick } from '../../hooks/useOuterCallback';
+import { Icon } from '../Icon';
+import { Padbox } from '../layout';
+import { H3 } from '../typography';
+import { getBorderRadius, getColor, getFontSize, pxToRem } from '../../utils';
+import { mergeRefs } from '../../utils/mergeRefs';
+import { SpaceSizes } from '../../theme';
+import { DSContext } from '../../theme/DSProvider/DSProvider';
+import { IconTypes, SSCIconNames } from '../../theme/icons/icons.enums';
+
+const widthVariants = {
+  [ModalWidthSizes.xs]: 320,
+  [ModalWidthSizes.sm]: 430,
+  [ModalWidthSizes.md]: 600,
+  [ModalWidthSizes.lg]: 900,
+};
+
+const Overlay = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  background: rgba(0, 0, 0, 0.6);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: 1250;
+`;
+
+const BaseModal = styled.div<{ $width: number }>`
+  position: relative;
+  width: ${({ $width }) => pxToRem($width)};
+  border-radius: ${getBorderRadius};
+  background-color: ${getColor('graphite5H')};
+`;
+
+const Title = styled(H3)`
+  margin: 0;
+`;
+
+// TODO Extract close button and unify it across DS
+const CloseButton = styled.button`
+  position: absolute;
+  top: ${pxToRem(16)};
+  right: ${pxToRem(16)};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: none;
+  flex-shrink: 0;
+  cursor: pointer;
+  color: ${getColor('graphite3B')};
+  width: ${pxToRem(16)};
+  height: ${pxToRem(16)};
+`;
+
+const TimesIcon = styled(Icon)`
+  font-size: ${getFontSize('lg')};
+`;
+
+const Content = styled(Padbox)<{ $maxHeight: number }>`
+  padding-top: 0;
+  overflow-y: auto;
+  max-height: ${({ $maxHeight }) =>
+    isNotNull($maxHeight) ? `${$maxHeight}px` : 'none'};
+`;
+
+const Footer = styled(Padbox)`
+  border-top: 1px solid ${getColor('graphiteHB')};
+`;
+
+const Modal: React.FC<ModalProps> = ({
```

Comment:
It would be great if all new components use forwardRef which is added to the base component, in this case BaseModal.
mergeRefs does support unlimited number of refs

Reply:
Got it, will bear it in mind